### ISSUE TITLE
0.2.0 minimal example documentation

### ISSUE THREAD

**Issue by jdkealy** [funcool/urania#13]
Hello! I'm evaluating 0.2.0 and couldn't get the minimal example working. It did work on 0.1.0. The example was from the 0.1.0 docs. Has something in the API changed since 0.1.0 ? 
```
(ns urania.testing
  (:require
   [urania.core :as u]
   [promesa.core :as prom]
   [clojure.set :refer [intersection]]))

(defn remote-req [id result]
  (prom/promise
   (fn [resolve reject]
      (let [wait (rand 1000)]
        (println (str "-->[ " id " ] waiting " wait))
        (Thread/sleep wait)
        (println (str "<--[ " id " ] finished, result: " result))
        (resolve result)))))

(defrecord FriendsOf [id]
  u/DataSource
  (-identity [_] id)
  (-fetch [_ _]
    (remote-req id (set (range id)))))

(defn friends-of [id]
  (FriendsOf. id))

(defn count-common
  [a b]
  (count (intersection a b)))

(defn count-common-friends
  [x y]
  (count-common (friends-of x) (friends-of y)))

(defn count-common-friends [x y]
  (u/map count-common
         (friends-of x)
         (friends-of y)))
(comment
  (deref (u/run! (friends-of 8))) ;; fn <!== this used to return the result set in 0.1.0

  ((u/run!! (friends-of 10)) nil nil) ;;  #{1,2,3,4,5,6,7,8,9,10}
  ((deref (u/run! (friends-of 10)) ) nil nil) ;;  #{1,2,3,4,5,6,7,8,9,10}
  )

```

**Comment by jdkealy** [comment]
I guess I didn't understand really what `(remote-req ` was doing. Replacing it with a promise and not with a promise that returns a functions seems to work. 
```
(defrecord FriendsOf [id]
  u/DataSource
  (-identity [_] id)
  (-fetch [_ _]
    (prom/promise (http/get "http://example.com" {:throw-exceptions false}))))
```

**Comment by oliyh** [comment]
This was due to the promesa upgrade. The documentation is fixed by #11 which I have just merged. Thanks for reporting!
